runtime-sdk/src/testing/mock.rs

```diff
@@ -232,14 +236,45 @@ impl Signer {
         C: Context,
         B: cbor::Encode,
     {
+        let inner_call = transaction::Call {
+            format: transaction::CallFormat::Plain,
+            method: method.to_owned(),
+            body: cbor::to_value(body),
+            ..Default::default()
+        };
+        let call = if !opts.encrypted {
+            inner_call
+        } else {
```

Comment:
It doesn't look like this is transparently handling encrypted results?

Reply:
It doesn't, but it calls `Dispatch::dispatch_tx`, which doesn't automatically encrypt the result object. I added a return value to both runtime methods in the access module's tests to check.